internal/querynode/data_sync_service.go

```diff
@@ -36,257 +36,144 @@ const (
 
-	if _, ok := dsService.collectionDeltaFlowGraphs[collectionID]; !ok {
-		dsService.collectionDeltaFlowGraphs[collectionID] = make(map[Channel]*queryNodeFlowGraph)
-	}
 	for _, vChannel := range vChannels {
-		// collection flow graph doesn't need partition id
-		partitionID := UniqueID(0)
+		if _, ok := dsService.deltaFlowGraphs[vChannel]; ok {
+			log.Warn("delta flow graph has been existed", zap.Any("Channel", vChannel))
+			continue
+		}
+		// delta flow graph doesn't need partition id
 		newFlowGraph := newQueryNodeDeltaFlowGraph(dsService.ctx,
-			loadTypeCollection,
 			collectionID,
-			partitionID,
 			dsService.historicalReplica,
 			dsService.tSafeReplica,
 			vChannel,
 			dsService.msFactory)
-		dsService.collectionDeltaFlowGraphs[collectionID][vChannel] = newFlowGraph
-		log.Debug("add collection flow graph",
+		dsService.deltaFlowGraphs[vChannel] = newFlowGraph
+		log.Debug("add delta flow graph",
 			zap.Any("collectionID", collectionID),
 			zap.Any("channel", vChannel))
 	}
 }
 
-// getCollectionFlowGraphs returns the collection flowGraph by collectionID
-func (dsService *dataSyncService) getCollectionFlowGraphs(collectionID UniqueID, vChannels []string) (map[Channel]*queryNodeFlowGraph, error) {
+// getDMLFlowGraph returns the DML flowGraph by collectionID
+func (dsService *dataSyncService) getDMLFlowGraph(collectionID UniqueID, channel Channel) (*queryNodeFlowGraph, error) {
 	dsService.mu.Lock()
 	defer dsService.mu.Unlock()
 
-	if _, ok := dsService.collectionFlowGraphs[collectionID]; !ok {
-		return nil, errors.New("collection flow graph doesn't existed, collectionID = " + fmt.Sprintln(collectionID))
-	}
-
-	tmpFGs := make(map[Channel]*queryNodeFlowGraph)
-	for _, channel := range vChannels {
-		if _, ok := dsService.collectionFlowGraphs[collectionID][channel]; ok {
-			tmpFGs[channel] = dsService.collectionFlowGraphs[collectionID][channel]
-		}
+	if _, ok := dsService.dmlFlowGraphs[channel]; !ok {
+		return nil, errors.New("DML flow graph doesn't existed, collectionID = " + fmt.Sprintln(collectionID))
 	}
 
-	return tmpFGs, nil
+	// TODO: return clone?
+	return dsService.dmlFlowGraphs[channel], nil
 }
 
-// getCollectionDeltaFlowGraphs returns the collection delta flowGraph by collectionID
-func (dsService *dataSyncService) getCollectionDeltaFlowGraphs(collectionID UniqueID, vChannels []string) (map[Channel]*queryNodeFlowGraph, error) {
+// getDeltaFlowGraph returns the delta flowGraph by collectionID
+func (dsService *dataSyncService) getDeltaFlowGraph(collectionID UniqueID, channel Channel) (*queryNodeFlowGraph, error) {
 	dsService.mu.Lock()
 	defer dsService.mu.Unlock()
 
-	if _, ok := dsService.collectionDeltaFlowGraphs[collectionID]; !ok {
-		return nil, errors.New("collection flow graph doesn't existed, collectionID = " + fmt.Sprintln(collectionID))
+	if _, ok := dsService.deltaFlowGraphs[channel]; !ok {
+		return nil, errors.New("delta flow graph doesn't existed, collectionID = " + fmt.Sprintln(collectionID))
```

Comment:
use fmt.Errorf(delta flow graph doesn't existed, collectionID = %d", collectionID) instead, don't not concatenate strings.

Reply:
fixed